presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -198,10 +204,88 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         CassandraTableHandle handle = checkType(table, CassandraTableHandle.class, "table");
         CassandraPartitionResult result = partitionManager.getPartitions(handle, constraint.getSummary());
-        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions()));
+        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions(), buildClusteringPredicate(schemaProvider.getTable(handle), constraint)));
         return ImmutableList.of(new ConnectorTableLayoutResult(layout, result.getUnenforcedConstraint()));
     }
 
+    private static Optional<String> buildClusteringPredicate(CassandraTable cassandraTable, Constraint<ColumnHandle> constraint)
+    {
+        List<CassandraColumnHandle> clusteringKeyColumns = cassandraTable.getClusteringKeyColumns();
+        Set<List<Object>> clusteringKeysSet = getClusteringKeysSet(cassandraTable, constraint.getSummary());
+        if (clusteringKeysSet == null || clusteringKeysSet.isEmpty()) {
+            return Optional.empty();
+        }
+
+        // todo Is it correct the way predicates are being appended to each other?
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int j = 0; j < clusteringKeysSet.size(); j++) {
+            List<Object> clusteringKeys = clusteringKeysSet.iterator().next();
```

Comment:
It will always return the very first element. I think that is not what you actually want to do. Once you change `getClusteringKeysSet` to return `List` - you can just do `clusteringKeysSet.get(j)`